torch/export/export_schema.py

```diff
@@ -0,0 +1,302 @@
+from dataclasses import dataclass
+from enum import auto, Enum
+from typing import List, Union, Dict
+
+
+################################################################################
+# Following section is the defining the permissible argument types for operators
```

Comment:
At a meta-level, I'm not sure if I want to define a general purpose serialization format (what this file is), or work on something more specific for whatever server side export case we're pushing, and then try to generalize from there.

Reply:
Could you elaborate on the requirements for server-specific export format? 

I am designing with an assumption that the serialized format must be easily consumable for a downstream C++ executor/runtime. (AFAIK, it's still unclear how the server side executor looks like, or do we have more clarity on how downstream system work now?) So my best effort, ATM, is to faithfully serialize the fx graph. 

cc @suo

Reply:
my 0.02: don't have opinions here except the first use case will be internal server definitely, so at least there's another option to develop all these internally first.

Reply:
> Could you elaborate on the requirements for server-specific export format?

I don't know what the requirements are. Figuring this out seems like the first order of business.

Reply:
My hunch is that it must be easily consumable by a downstream  *compiler*. I.e. if downstream chooses to implement a runtime, then, it likely also implement a compiler that takes the exported format and generate their own format. We'd like not to restrict what language the compiler is written, therefore preferring thrift or other serialization schema over pickle.

Examples of this usage pattern:
* Executorch is the runtime; exir capture / to_edge / to_backend / emit is the compiler
* GPU / CPU is the runtime (native); torchinductor + triton / gcc is the compiler.

Reply:
> must be easily consumable by a downstream compiler.

I think this is the right goal. This is not meant to be used as-is for execution, which means it's not meant to be concerned with performance, efficiency, portability, ... at least to the level that an executable format should be.

This format would also be used as a transport mechanism for sharing models around that can be then targeted to 1+ executors. E.g .this is what a product-research team would save and pass to the ML-engineering team to further target and optimize.